src/pages/Overview/OverviewPage.tsx

```diff
@@ -99,76 +88,100 @@ class OverviewPage extends React.Component<OverviewProps, State> {
-        namespace: namespace,
-        health: r
-      }));
+    // debounce async for back-pressure, ten by ten
+    _.chunk(this.state.namespaces, 10).forEach(chunk => {
+      this.promises
+        .registerChained('healthchunks', undefined, () => this.fetchHealthChunk(chunk, rateInterval, type))
+        .then(() => {
+          this.setState(prevState => {
+            let newNamespaces = prevState.namespaces.slice();
+            if (sortField.id === 'health') {
+              newNamespaces = FiltersAndSorts.sortFunc(newNamespaces, sortField, isAscending);
+            }
+            return { namespaces: newNamespaces };
+          });
+        });
     });
-    Promise.all(promises)
-      .then(responses => {
-        const allNamespaces: NamespaceInfo[] = [];
-        responses.forEach(response => {
-          const info: NamespaceInfo = {
-            name: response.namespace,
+  }
+
+  fetchHealthChunk(chunk: NamespaceInfo[], rateInterval: number, type: OverviewType) {
+    const apiFunc = switchType(
+      type,
+      API.getNamespaceAppHealth,
+      API.getNamespaceServiceHealth,
+      API.getNamespaceWorkloadHealth
+    );
+    return Promise.all(
+      chunk.map(nsInfo => {
+        const healthPromise: Promise<NamespaceAppHealth | NamespaceWorkloadHealth | NamespaceServiceHealth> = apiFunc(
+          authentication(),
+          nsInfo.name,
+          rateInterval
+        );
+        return healthPromise.then(rs => ({ health: rs, nsInfo: nsInfo }));
+      })
+    )
+      .then(results => {
+        results.forEach(result => {
+          const nsStatus: NamespaceStatus = {
             inError: [],
             inWarning: [],
             inSuccess: [],
             notAvailable: []
           };
-          const { showInError, showInWarning, showInSuccess, noFilter } = OverviewPage.summarizeHealthFilters();
-          let show = noFilter;
-          Object.keys(response.health).forEach(item => {
-            const health: Health = response.health[item];
+          Object.keys(result.health).forEach(item => {
+            const health: Health = result.health[item];
             const status = health.getGlobalStatus();
             if (status === FAILURE) {
-              info.inError.push(item);
-              show = show || showInError;
+              nsStatus.inError.push(item);
             } else if (status === DEGRADED) {
-              info.inWarning.push(item);
-              show = show || showInWarning;
+              nsStatus.inWarning.push(item);
             } else if (status === HEALTHY) {
-              info.inSuccess.push(item);
-              show = show || showInSuccess;
+              nsStatus.inSuccess.push(item);
             } else {
-              info.notAvailable.push(item);
+              nsStatus.notAvailable.push(item);
             }
           });
-          if (show) {
-            allNamespaces.push(info);
-          }
+          result.nsInfo.status = nsStatus;
```

Comment:
It is on purpose that the state isn't set at this point, because this is after a promise that is itself wrapped into a cancelable. If we update state here and the wrapping promise is cancelled, the "setstate" would still occur.